index.js

```diff
@@ -94,14 +101,22 @@ class ServiceClientFactory {
     });
   }
 
-  createServiceClient(serviceName, url, audience, tokenRefreshRate) {
+  createServiceClient(
+    serviceName,
+    url,
+    audience,
+    tokenRefreshRate,
+    getCorrelationId,
+  ) {
     if (!this.initialized) {
       throw new Error('Cannot create client in uninitialized ServiceClientFactory');
     }
 
+    getCorrelationId = getCorrelationId || function cid() {};
```

Comment:
Why does this need to be a function and not a string? couldn't the caller async get the id first?

Reply:
Also spoke in IRL: instead of overwriting the arg, it should default to a utility `noop` function.

Reply:
It's a function because the correlation id changes for each request to the BFF.